### ISSUE TITLE
API does not work in Cross domain request - No access allowed from other domain 

### ISSUE THREAD

**Issue by cnngraphics** [ridwanskaterock/cicool#191]
What is the point of having an api if it does not work outside of same domain?
I have been trying for 2 weeks to see if I can have my localhost work with cicool installed on a web server
no luckz!!!!!!!!!!!!!!

PostMan Works, but nothing else work

![image](https://user-images.githubusercontent.com/1514378/41227236-28e16ee8-6d42-11e8-838a-c828f8bf7926.png)

**Comment by TEX-Team** [comment]
same problem too

**Comment by cnngraphics** [comment]
@tex-team
I am still fighting with this issue today... no answer from the creator of CICool for days now. I wonder if he knows what CORS is?

**Comment by ridwanskaterock** [comment]
may know I've used cicool by integrating the API on more than 10 websites, and it works fine,

please follow the documentation with the appropriate METHOD and also part of the security such as X-Api-Key and X-Token

and I have mencopa on another server to access the API by using CURL, Guzzle HTTP library, and also on the client can use JQUERY for it and work well

**Comment by cnngraphics** [comment]
No more CORS Issues, but now this:

![image](https://user-images.githubusercontent.com/1514378/41319810-10da2e3c-6e6b-11e8-91bc-cb1f5729b74b.png)

**Comment by cnngraphics** [comment]
The solution to disable CORS in Chrome is a bad one because: 
If I deliver a Web ap to another user and on their server or inside of a phone, they won't be able to disable this chrome CORS, this is a SERVER issue that need to be addressed from CICOOL settings and code.
If this is solved that would a master achievement.

Also, your many installation that work are from same server to same server calls. (SAME ORIGIN) that is why there is no issue.

Also this tells me you need to learn more about CORS issues. 
I need to leanr more about this too. If I find a solution I will share it here.

But CORS issues only show when calling an API domain from

**Comment by ridwanskaterock** [comment]
can you add some code like that




on application/core/My_Controller.php


```php
$this->output->set_header("Access-Control-Allow-Origin: *");
 $this->output->set_header("Access-Control-Allow-Methods: GET, POST,
OPTIONS, PUT, DELETE");
```

2018-06-13 5:11 GMT+07:00 Theo M. <notifications@github.com>:

**Comment by ridwanskaterock** [comment]
or you can try change value configuration for file 

`application/config/rest.php `

![image](https://user-images.githubusercontent.com/9270445/41320354-acc019ac-6ec9-11e8-889b-f8b53fd2b008.png)

or 
set to TRUE 

![image](https://user-images.githubusercontent.com/9270445/41320367-b851ee12-6ec9-11e8-9761-be377f7a2037.png)

**Comment by cnngraphics** [comment]
i tried what you just said about rest.php file. The issue is the same.
I event tried to upload the app to the serve to test and I got what you see in the image below.
I got two different errors: 403 and 406
See below urls and scfreenshot.

http://cicool.proappstore.com/www/ (is the same domain as cicool installation)
http://proappstore.com/src/ (Different domain) 

![image](https://user-images.githubusercontent.com/1514378/41327041-e9a9875c-6e8f-11e8-8f1a-aa5ac23331e3.png)
